server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/GetMessageListMethod.java

```diff
@@ -161,6 +176,37 @@ private MultimailboxesSearchQuery convertToSearchQuery(GetMessageListRequest mes
                 .build();
     }
 
+    private boolean containsNestedMailboxFilters(Filter filter) {
+        if (filter instanceof FilterOperator) {
+            FilterOperator operator = (FilterOperator) filter;
+
+            return operator.getConditions()
+                .stream()
+                .anyMatch(this::containsMailboxFilters);
+        }
+        if (filter instanceof FilterCondition) {
+            // The condition is not nested
+            return false;
+        }
+        throw new RuntimeException("Unsupported Filter implementation " + filter);
+    }
+
+    private boolean containsMailboxFilters(Filter filter) {
+        if (filter instanceof FilterOperator) {
+            FilterOperator operator = (FilterOperator) filter;
+
+            return operator.getConditions()
+                .stream()
+                .anyMatch(this::containsMailboxFilters);
```

Comment:
what happens in case of too many levels of nested filters? Does the server crash?
I think we should add a test about this, because as the filters can be easily forged by the client, we should not add an attack vector here.

Reply:
Let's allow maximum 10 nested levels then ;-)